### PR TITLE
Fixed RichTextLabel wrong selection offset after drop cap

### PR DESCRIPTION
**Pull request by Koyper**
Fixes an issue in RichTextLabel where starting a selection, or selecting across a drop cap, would incorrectly start the selection offset by the width of the drop cap.

I would recommend leaving in the comments, because this is where a future improvement would fix the issue of the drop cap text not currently being selectable, and figuring out how to obtain the drop cap text is not obvious.

### CODE REVIEW

**Comment by YuriSizov** [comment]
> I would recommend leaving in the comments, because this is where a future improvement would fix the issue of the drop cap text not currently being selectable, and figuring out how to obtain the drop cap text is not obvious.

If this code is not valid yet, then there is no point leaving it there. You can stash the changes and work on it after we merge this. So please strip these lines from the PR.

File: scene/gui/rich_text_label.cpp [hunk 1]
```diff
@@ -1470,7 +1470,32 @@ float RichTextLabel::_find_click_in_line(ItemFrame *p_frame, int p_line, const V
...
+			// 			}
+			// 		}
+			// 		print_line("DC TEXT", drop->text);
+			// 	}
+			// }
+			if (rtl) {
+				off.x -= h_off;
+			} else {
+				off.x += h_off;
+			}
```
Comment (bruvzg) [inline scene/gui/rich_text_label.cpp:1497]: Drop cap offset change is correct.

The selection text code should probably be done as a separate PR. This will need to do hit_test on drop cap text buffer and set frame/item/char offset accordingly, and might need some extra changes (I do not remember if drop cap counts towards char offset at all).

**Comment by YuriSizov** [comment]
You need to squash your commits.

**Comment by Koyper** [comment]
>If this code is not valid yet, then there is no point leaving it there. You can stash the changes and work on it after we merge this. So please strip these lines from the PR.

Is there a mechanism somewhere for adding code comments like this one for the benefit of a different, future contributor? I'll have my own comments, but it could save someone else some effort?

**Comment by YuriSizov** [comment]
> Is there a mechanism somewhere for adding code comments like this one for the benefit of a different, future contributor? I'll have my own comments, but it could save someone else some effort?

You can comment your suggestion on an issue. If there is no issue, you can open one and leave a comment there. Code comments aren't discoverable and having dead code in the codebase is mostly just contributes to the noise.

**Review by YuriSizov — APPROVED** [review]
LGTM. bruvzg approved, commented out lines were removed.

**Comment by YuriSizov** [comment]
Thanks!

**Comment by Koyper** [comment]
>Code comments aren't discoverable and having dead code in the codebase is mostly just contributes to the noise. And you also cannot guarantee that your solution is the correct one, so leaving it in the code would be misleading.

I agree with these good observations - thanks!
